zerver/views/streams.py

```diff
@@ -149,6 +148,15 @@ def update_stream_backend(
         new_name: Optional[Text]=REQ(validator=check_string, default=None),
 ) -> HttpResponse:
     (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id)
+    # Only realm admin and stream admin can update stream properties
+    # Realm admin can update unsubscribed public stream property but can not
+    # update unsubscribed private stream property.
+    if sub is None:
+        if not stream.invite_only and not user_profile.is_realm_admin:
+            return json_error(_("Invalid stream id"))
+    else:
+        if not (user_profile.is_realm_admin or sub.is_admin):
```

Comment:
I'd add a comment here "If a realm admin is subscribed to a stream, they get admin powers for that stream."